nmoscommon/mdns/dnsUtils.py

```diff
@@ -15,12 +15,16 @@
 # limitations under the License.
 
 import dns.resolver  # Importing dns alone is not enough for NXDOMAIN to work
-from nmoscommon.mdns.mdnsExceptions import DNSRecordNotFound
+from nmoscommon.mdns.mdnsExceptions import DNSRecordNotFound, NoDefaultDnsSearchDomian
 
 
 def _defaultDomain():
-    resolve = dns.resolver.Resolver()
-    return str(resolve.search[0])
+    try:
+        resolve = dns.resolver.Resolver()
+        return str(resolve.search[0])
+    except IndexError: 
+        print("No default search domain found in /etc/resolv.conf")
```

Comment:
Could this be removed? The log statement when the exception is caught should be sufficient.